pkg/gridtypes/zos/gw_fqdn.go

```diff
@@ -19,24 +20,41 @@ type Backend string
 
 // check if valid http://ip:port, http://ip or ip:port
 func (b Backend) Valid(tls bool) error {
-	u, err := url.Parse(string(b))
-	if err != nil {
-		return errors.Wrap(err, "failed to parse backend")
-	}
+	var hostName string
 	if tls {
-		if u.Scheme != "" {
-			return fmt.Errorf("scheme expected to be empty")
+		host, port, err := net.SplitHostPort(string(b))
+		if err != nil {
+			return fmt.Errorf("failed to parse backend with error: %w", err)
+		}
+		if port == "" {
+			return fmt.Errorf("missing port in backend")
 		}
-		if u.Port() == "" {
-			return fmt.Errorf("missing port in backend address")
+
+		parsedPort, err := strconv.ParseInt(port, 10, 64)
+		if err != nil {
+			return fmt.Errorf("invalid port in backend: %s", port)
+		}
+
+		if parsedPort > 65535 || parsedPort < 0 {
```

Comment:
flip the OR'd items to be parsedPort < 0 || parsedPort > 65535

Reply:
Once possiblity of negative value is omitted, she can only validate the value is less than or equal `math.MaxUint16` please use constants, not hardcoded values